sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorHandlerProvider.java

```diff
@@ -143,6 +164,21 @@ public SendLinkHandler createSendLinkHandler(String connectionId, String hostnam
     public ReceiveLinkHandler createReceiveLinkHandler(String connectionId, String hostname,
         String receiverName, String entityPath) {
 
-        return new ReceiveLinkHandler(connectionId, hostname, receiverName, entityPath);
+        return new ReceiveLinkHandler(connectionId, hostname, receiverName, entityPath, getMetricProvider(hostname, entityPath));
+    }
+
+    /**
+     * Returns cached {@link AmqpMetricsProvider} (or creates one) for given meter and entity.
+     * It's recommended to keep returned value in instance variable and to avoid calling
+     * this method extensively.
+     */
+    AmqpMetricsProvider getMetricProvider(String namespace, String entityPath) {
+        if (meter != null && !meter.isEnabled()) {
+            return AmqpMetricsProvider.noop();
+        }
+
+        return metricsCache.computeIfAbsent(
+            namespace + (entityPath == null ? "" : "/" + entityPath),
+            ignored -> new AmqpMetricsProvider(meter, namespace, entityPath));
```

Comment:
I'm curious what the expectation is in this method if the Meter given in the constructor is null?

Reply:
if meter is null, `AmqpMetricsProvider` uses default meter (i.e. creates one using default options).

The next step would be creating meter in EH/SB and plumbing it through, to AMQP core so the default meter will never be used and users will be able to configure metrics if they want.

I'm just phasing this work across several releases to reduce dependency mismatch pain with the new EB/EH + old amqp core.